tests/unit/materializers/test_built_in_materializer.py

```diff
@@ -129,3 +135,81 @@ def test_none_values():
             step_output_type=type_, step_output=example
         )
         assert result == example
+
+
+class CustomType:
+    """Custom type used for testing the container materializer below."""
+
+    myname = "aria"
+
+    def __eq__(self, __value: object) -> bool:
+        if isinstance(__value, CustomType):
+            return self.myname == __value.myname
+        return False
+
+
+class CustomSubType(CustomType):
+    """Subtype of CustomType."""
+
+    myname = "axl"
+
+
+class CustomTypeMaterializer(BaseMaterializer):
+    """Mock materializer for custom types.
+
+    Does not actually save anything to disk, just initializes the type.
+    """
+
+    ASSOCIATED_TYPES = (CustomType,)
+
+    def save(self, data: CustomType) -> None:
+        """Save the data (not)."""
+        pass
+
+    def load(self, data_type: Type[CustomType]) -> Optional[CustomType]:
+        """Load the data."""
+        return data_type()
+
+
+def test_container_materializer_for_custom_types(mocker):
+    """Test container materializer for custom types.
+
+    This ensures that:
+    - The container materializer can handle custom types.
+    - Custom types are loaded as the correct type.
+    - The materializer of the subtype does not need to be registered in the
+        materializer registry when the container is loaded.
+    """
+    from zenml.materializers.default_materializer_registry import (
+        default_materializer_registry,
+    )
+
+    example = [CustomType(), CustomSubType()]
+    with TemporaryDirectory() as artifact_uri:
+        materializer = BuiltInContainerMaterializer(uri=artifact_uri)
+
+        # Container materializer should find materializer for both elements in
+        # the default materializer registry.
+        mocker.patch.object(
```

Comment:
Shouldn't this be in the registry anyway as we inherit from `BaseMaterializer` with the associated type set?

Reply:
Yeah, true, it also works without this.